components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java

```diff
@@ -268,9 +268,10 @@ private SecretAddRequest buildSecretAddFromSecretUpdateRequest(String name,
     private APIError handleSecretMgtException(SecretManagementException e, SecretManagementConstants.ErrorMessage
             errorEnum, String data) {
 
-        ErrorResponse errorResponse = getErrorBuilder(errorEnum, data).build(log, e, errorEnum.getDescription());
+        ErrorResponse errorResponse;
```

Comment:
why we moved this logic into each if claus? 
`errorResponse = getErrorBuilder(errorEnum, data).build(log, e.getMessage());`
Since this is the way we use in all the places, we have have it in the top (common to if and else condition)

Reply:
otherwise it will create logs for all the exceptions. With the modification it will create logs for server exceptions and only debug logs for client exceptions.